calc/rule_scanner.go

```diff
@@ -16,21 +16,23 @@ package calc
 import (
 	log "github.com/Sirupsen/logrus"
 
+	"fmt"
+
 	"github.com/projectcalico/felix/multidict"
 	"github.com/projectcalico/felix/set"
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
-	"github.com/projectcalico/libcalico-go/lib/hash"
 	"github.com/projectcalico/libcalico-go/lib/net"
 	"github.com/projectcalico/libcalico-go/lib/numorstring"
 	"github.com/projectcalico/libcalico-go/lib/selector"
 )
 
-// RuleScanner calculates which selectors and tags are in use by the active rules (calculated
-// by the ActiveRulesCalculator).  I.e. which tags and selectors are in the rules and need to
-// be rendered to the dataplane (as IP sets, for example).
+// RuleScanner scans the rules sent to it by the ActiveRulesCalculator, looking for tags and
+// selectors. It calculates the set of active tags and selectors and emits events when they become
+// active/inactive.  To avoid the occupancy overhead of doing two types of indexing, it maps tags
+// to selectors of the form "has(tag-name)"; those are broadly equivalent due to the inheritance
```

Comment:
'broadly' doesn't sound very reassuring here.  Can this say something more definitive?